ui-episodedetails/src/main/java/app/tivi/episodedetails/EpisodeDetailsViewModel.kt

```diff
@@ -55,29 +56,38 @@ internal class EpisodeDetailsViewModel @Inject constructor(
     private val removeEpisodeWatch: RemoveEpisodeWatch,
     private val logger: Logger,
     private val snackbarManager: SnackbarManager
-) : ReduxViewModel<EpisodeDetailsViewState>(
-    EpisodeDetailsViewState(
-        // The string "episodeId" is the name of the argument in the route
-        episodeId = savedStateHandle.get("episodeId")!!
-    )
-) {
+) : ViewModel() {
+
+    private val episodeId: Long = savedStateHandle.get("episodeId")!!
 
     private val loadingState = ObservableLoadingCounter()
 
     private val pendingActions = MutableSharedFlow<EpisodeDetailsAction>()
 
-    init {
-        viewModelScope.launch {
-            observeEpisodeDetails.observe()
-                .collect { updateFromEpisodeDetails(it) }
-        }
-
-        viewModelScope.launch {
-            observeEpisodeWatches.observe()
-                .onStart { emit(emptyList()) }
-                .collect { updateFromEpisodeWatches(it) }
-        }
+    val state = combine(
+        observeEpisodeDetails.observe(),
+        observeEpisodeWatches.observe(),
+        loadingState.observable,
+        snackbarManager.flow.onStart { emit(null) },
```

Comment:
Why do you need to emit `null` on start? `snackbarManager.flow` already does that for you, right?

Reply:
Good call, removed.